File: app/1DSSHBerryPhase.cpp
#include <iostream>
#include "matplotlibcpp.h"
#include <armadillo>
#include "SSH_1D.h"

using namespace std;
using namespace arma;
namespace plt = matplotlibcpp;

int main()
{
  int n = 100;
  int N = 10;
  vector<double> x(n), y(n);

  double t1 = 1;
  double t2 = 2;

  SSH_1D mod(t1,t2);

  for(int i = 0; i < n; i++){
    x[i] = 2*(double)i/n;
    mod.set_interhop(x[i]);
    y[i] = mod.berry_kspace(N); 
  }

  plt::plot(x,y);
  plt::xlabel("J'");
  plt::ylabel("$ \\phi$");
  plt::title("Berry phase, J = 1");
  plt::show();

  return 0;
}

File: scripts/1DSSHBerryPhase.py
import helper as hp
import matplotlib.pyplot as plt

name = "1DSSHBerryPhase"
data = hp.readfile(name + ".dat")

fig, ax = plt.subplots()
ax.plot(data[0],data[1], color = 'blue')

ax.set(xlabel = r'$J\prime$', ylabel = r'$\phi$', title=r'Berry phase, $J = 1$')

fig.savefig(hp.plot_dir() + name + ".png", dpi = 200)
plt.show()
